toolkit/tools/pkg/imagecustomizerlib/customizeboot.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+)
+
+var (
+	linuxCommandLineRegex = regexp.MustCompile(`\tlinux .* (\$kernelopts)`)
+)
+
+func handleKernelCommandLine(commandLineAdd string, imageChroot *safechroot.Chroot) error {
+	var err error
+
+	if commandLineAdd == "" {
+		// Nothing to do.
+		return nil
+	}
+
+	grub2ConfigFilePath := filepath.Join(imageChroot.RootDir(), "/boot/grub2/grub.cfg")
+
+	// Read the existing grub.cfg file.
+	grub2ConfigFileBytes, err := os.ReadFile(grub2ConfigFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read existing grub2 config file: %w", err)
+	}
+
+	grub2ConfigFile := string(grub2ConfigFileBytes)
+
+	// Find the point where the new command line arguments should be added.
+	match := linuxCommandLineRegex.FindStringSubmatchIndex(grub2ConfigFile)
+	if match == nil {
+		return fmt.Errorf("failed to find Linux kernel command line params in grub2 config file")
+	}
+
+	// Get the location of "$kernelopts".
```

Comment:
Splicing directly into /boot/grub2/grub.cfg will not be persistent. We need to find a way to edit /etc/default/grub, /etc/default/grub.d/*.cfg in order for mkconfig to be supported.

Reply:
Yeah, I have a separate task for handling `grub-mkconfig` enabled images.